refmt.el

```diff
@@ -201,8 +203,19 @@ function."
            (if (zerop (let* ((files (list (list :file outputfile) errorfile))
                              (args (append width-args (list "--parse" from "--print" to bufferfile))))
                         (cond ((equal refmt-command 'opam)
-                               (apply 'call-process
-                                      "opam" nil files nil (append '("exec" "--" "refmt") args)))
+                               (if refmt-opam-bin-dir
+                                   (apply 'call-process (concat refmt-opam-bin-dir "/refmt") nil files nil args)
+
+                                 (progn
+                                   (setq-local refmt-opam-bin-dir (with-temp-buffer
+                                                                    (when (eq (call-process-shell-command
+                                                                               "opam config var bin" nil (current-buffer) nil) 0)
+                                                                      (let ((bin-path
+                                                                             (replace-regexp-in-string "\n$" "" (buffer-string))))
+                                                                        ;; this was originally done via `opam exec' but that does not
+                                                                        ;; work for opam 1, and added a performance hit
+                                                                        bin-path))))
+                                   (apply 'call-process (concat refmt-opam-bin-dir "/refmt") nil files nil args))))
```

Comment:
I would refactor this code a bit so that there is only one occurrence of `(apply 'call-process (concat refmt-opam-bin-dir "/refmt") nil files nil args)`